File: 2sum.cpp
#include<bits/stdc++.h>
using namespace std;
int n;
bool found(int arr[], int k){
    for(int i=0;i<n;i++){
        if(arr[i]==k){
            return true;
        }
    }
    return false;
}
bool twosum(int arr[],int k){
    for(int i=0;i<n;i++){
        int num=arr[i];
        for(int j=0;j<n;j++){
            if(found(arr,(k-num))){
                return true;
            }
        }
    }
    return false;

}
int main(){
    int n;
    cin>>n;
    int k;
    cin>>k;
    int arr[n];
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }
    if(twosum(arr,k)){
        cout<<"the two sum exit in the array ";
    }
    else{
        cout<<"The two not exist in the array ";
    }
    return 0;

}

File: 2sum.py
import sys


def two_sum_exists(nums, target):
    seen = set()
    for num in nums:
        if target - num in seen:
            return True
        seen.add(num)
    return False


def main():
    data = list(map(int, sys.stdin.read().strip().split()))
    if not data:
        return
    if len(data) < 2:
        print("The two not exist in the array ")
        return
    n, target, *rest = data
    nums = rest[:n]
    if two_sum_exists(nums, target):
        print("the two sum exit in the array ")
    else:
        print("The two not exist in the array ")


if __name__ == "__main__":
    main()
